packit_service/service/api/runs.py

```diff
@@ -36,20 +36,25 @@ def process_runs(runs):
     result = []
 
     for pipeline in runs:
```

Comment:
just a heads up it's really temporary fix, cause you get already grouped by rows in `runs` ⇒ if you have two different projects triggering test without build in a row, they will get grouped together…

---

HOWEVER! Now that I'm thinking about it, we have `job_trigger` relation too, so it would probably make more sense to group them by the `job_trigger_id` (that must be set even for tests-only runs, right?). So that would resolve our issue with empty SRPM field.

`<already done in PR, I should try look ahead sometimes>`
And now we just have to decide where to get the `"trigger"` key from, but that is easy, since it works for all models (SRPM, Copr, Koji, TF), so it's just workflow like:

```
FOREACH job_type OF { copr, koji, testing_farm } DO
  FOREACH packit_id OF … DO
    // yada-yada-yada we do already
    IF trigger NOT IN response_dict THEN
      response_dict[trigger] <- get_project_info_from_build(row)
    FI
  OD
OD
```

and it also sounds like Friday work, since `get_project_info_from_build` supports testing farm too :upside_down_face:

`</already done in PR, I should try look ahead sometimes>`